api/tests/test_rate_limit.py

```diff
@@ -334,6 +334,31 @@ async def test_refresh_server_liveness(
             await live_server.acquire_connection('user-1', 'room-last')
 
 
+@any_rate_limiter_factory
+async def test_refresh_server_liveness_many_users(
+    rate_limiter_factory: RateLimiterFactory,
+) -> None:
+    with time_machine.travel('1970-01-01', tick=False) as traveller:
+        refreshing_server = await rate_limiter_factory('server-id-1')
+
+        users = [f'user-{i}' for i in range(50)]
+        for i, user in enumerate(users):
+            await refreshing_server.acquire_connection(user, f'room-{i}')
+
+        for i in range(MAX_CONNECTIONS_PER_USER):
+            await refreshing_server.acquire_connection('greedy-user', f'room-{i}')
```

Comment:
Is there some edge case in server refreshing when a user has a lot of connections? It's not clear to me why this is done in this test.

Reply:
No, but I need for a user to have enough connections that the final one we try to add at the bottom causes an exception to be thrown for too many connections

Reply:
That's part of the thing I'm not clear on. Rejecting a connection because a user has too many connections also seems unrelated to refreshing a server with many users.

Reply:
We need to prove that the refresh worked, i.e. those users still count as active connections. So we refresh, wait until they would expire if they weren't refreshed, and finally try to add another connection. If the refresh failed the connection would be allowed, but since it succeeded it should throw an exception when we try to add another connection for that user